lib/topological_inventory/ansible_tower/collector.rb

```diff
@@ -68,8 +86,15 @@ def collector_thread(connection, entity_type)
       sweep_scope = Set.new
       cnt = 0
       refresh_state_part_collected_at = nil
-      # each on ansible_tower_client's enumeration makes pagination requests by itself
-      send("get_#{entity_type}", connection).each do |entity|
+
+      heartbeat_thread_method = standalone_mode ? :run_in_parallel_with : :run_thread_queue_in_parallel_with
+      entities = heartbeat_queue.send(heartbeat_thread_method) do
+                   send("get_#{entity_type}", connection)
```

Comment:
what's the real timeout there? Is it open_timeout + read_timeout or the read_timeout includes the open_timeout?
Just thinking that the heartbeat timeout is set to 65s and open_timeout + (default) read_timeout is 70s

Reply:
it will 65 seconds keep touching heartbeat file(each 10 seconds.) so last time when file was touched is on 60th seconds, and heartbeat check time out is set to 12 seconds.
so basically heartbeat check will start to fail(return value other than 0) after 72seconds when request started.

According to my testing read_timeout includes the open_timeout.